rust/windows-client/src-tauri/src/client/auth.rs

```diff
@@ -1,21 +1,34 @@
 //! Fulfills <https://github.com/firezone/firezone/issues/2823>
 
-use connlib_shared::control::SecureUrl;
+use connlib_shared::{control::SecureUrl, windows::app_local_data_dir};
 use rand::{thread_rng, RngCore};
 use secrecy::{ExposeSecret, Secret, SecretString};
+use std::path::PathBuf;
 use subtle::ConstantTimeEq;
 use url::Url;
 
 const NONCE_LENGTH: usize = 32;
 
 #[derive(thiserror::Error, Debug)]
 pub(crate) enum Error {
+    #[error("`actor_name_path` has no parent, this should be impossible")]
+    ActorNamePathWrong,
+    #[error("`app_local_data_dir` failed")]
+    CantFindLocalAppDataFolder,
+    #[error("`create_dir_all` failed while writing `actor_name_path`")]
+    CreateDirAll(std::io::Error),
+    #[error("Couldn't delete actor_name from disk: {0}")]
+    DeleteActorName(std::io::Error),
```

Comment:
Now that this is growing, is there any reason we aren't using `anyhow` considering this is a binary crate?

Reply:
Also, although I've never seen `anyhow` crash, it does use unsafe internally and `thiserror` does not, last time I checked

Reply:
> Also, although I've never seen anyhow crash, it does use unsafe internally and thiserror does not, last time I checked

`anyhow` is one of the most widely used crates, we can assume it's safe

> I like that with thiserror none of the Froms are implemented by default, so I can't just put ? and forget to add a .context

With anyhow you can have precise control over the context given the code-path and I'm not sure if the overhead of maintaining this enum is worth having this automatically added.